src/ImageSharp/Memory/Allocators/Internals/Gen2GcCallback.cs

```diff
@@ -43,7 +42,7 @@ private Gen2GcCallback(Func<object, bool> callback, object targetObj)
             // Execute the callback method.
             try
             {
-                if (!this.callback1(targetObj))
+                if (this.callback1 is not null && !this.callback1(targetObj))
```

Comment:
`callback1`should never be null here. The type was ported from [here](https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/Gen2GcCallback.cs). We don't use `Debug.Assert` in our codebase, so this should be probably

```suggestion
                if (!this.callback1!(targetObj))
```

Reply:
callback1 or callback0 can be null. It depends on the used constructor. It will just be a silent exception. I can replace it with the !

Reply:
The point is that if you add stuff like `this.callback1 is not null`, or turn `this.someField.Call()` to `this.someField?.Call() there no error will occur if a bug is introduced to the code later. We should be very careful with such changes, better not to make them unless it's 100% safe and future-proof.